### ISSUE TITLE
Unable to use connection pooling

### ISSUE THREAD

**Issue by Yakumo-Yukari** [moigagoo/norm#206]
```
import norm/[model, sqlite, pool]
import user


proc myDb(): DbConn = open("mydb.db", "", "", "")


var connPool* = newPool[DbConn](10,myDb)

proc initDb*() =
  withDb(connPool):
    db.createTables(newUser())

proc resetPool*() =
  connPool.reset()

proc closePool*() =
  close(connPool)
```
error message

`C:\Users\.nimble\pkgs2\norm-2.8.3-c79c7ce80ebc2264679d6a252b0250ed3b4bd867\norm\pool.nim(20, 70) Error: type mismatch: got 'None' for 'getDb' but expected 'proc (): T: DbConn{.closure.}'`

**Comment by PhilippMDoerner** [comment]
The nim version may be relevant here as this compiles under nim 2.0.8

**Comment by moigagoo** [comment]
This looks so weird. Nim doesn't even have a built-in `None` value as far as I know.

@PhilippMDoerner I'm trying to look into it. Could you please take a look as well? I'm completely confused right now.

The issue can be reproduced with `testament r tests/sqlite/tpool.nim`.

**Comment by moigagoo** [comment]
The test code doesn't matter though. All you need is `import norm/[sqlite, pool]` and the compilation will fail.

**Comment by moigagoo** [comment]
OK it seems like in Nim 2.2.0 `getDb` can no longer be resolved to `sqlite.getDb` or `postgres.getDb` based on the type of DbConn.

So this code is no longer valid:
```nim
proc newPool*[T: DbConn](defaultSize: Positive, getDbProc: proc: T = getDb, poolExhaustedPolicy = pepRaise): Pool[T] =
```

We now have to have two `newPool` procs: one with explicit `sqlite.getDb`, the other with explicit `postgres.getDb`.

The CI got broken, I'll need to fix that too.

**Comment by moigagoo** [comment]
@Yakumo-Yukari the issue has been fixed in 2.8.4.

The new version is to be released shortly.

Thank you for reporting the issue 🙏
